<a href="https://colab.research.google.com/github/Gaukhar-ai/for_my_Thinkful_work/blob/master/dask_array.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Solutions to checkpoint 4

In the following exercises, you'll be working with the code snippet below:

```python
%%timeit
x = da.random.random((10000, 10000), chunks=(1000, 1000))
y = x + x.T
z = y[::2, 5000:].mean(axis=1)
z.compute()
```

Please submit your solutions to the following tasks as a link to your jupyter notebook on Github.

In [1]:
import dask.array as da

## 1. Change the code above by setting `chunks=(250, 250)`. How long does it take to run?

In [2]:
%%timeit
x = da.random.random((10000, 10000), chunks=(250, 250))
y = x + x.T
z = y[::2, 5000:].mean(axis=1)
z.compute()

1 loop, best of 3: 1.88 s per loop


It took around 1.88 seconds or 1230 milliseconds.

*   List item
*   List item



## 2. Now, set the parameter to `chunks=(500, 500)`. How long does it take to run? Does this one or the previous one run quickly? Why?

In [3]:
%%timeit
x = da.random.random((10000, 10000), chunks=(500, 500))
y = x + x.T
z = y[::2, 5000:].mean(axis=1)
z.compute()

1 loop, best of 3: 1.08 s per loop


It took around 1.08 milliseconds which is around half of what it takes in the previous task. This is because, when we divide a Dask array into smaller chunks Dask needs to manage a higher number of chunks. This requires more computation. If our dataset can fit into memory as is in our case, then setting the chunk size to a larger number speeds up the computation.